zipkin-common/src/test/scala/com/twitter/zipkin/storage/DependencyStoreSpec.scala

```diff
@@ -188,6 +188,51 @@ abstract class DependencyStoreSpec extends JUnitSuite with Matchers {
     result(store.getDependencies(today + 2 * day, Some(day))) should be(empty)
   }
 
+  @Test def getDependencies_intermediateSpans() = {
+    val day = MILLISECONDS.convert(1, DAYS)
+    val today = Time.now.floor(Duration.fromMilliseconds(day)).inMillis
+    val zipkinWeb = Endpoint(172 << 24 | 17 << 16 | 3, 8080, "zipkin-web")
+    val zipkinQuery = Endpoint(172 << 24 | 17 << 16 | 2, 9411, "zipkin-query")
+    val zipkinJdbc = Endpoint(172 << 24 | 17 << 16 | 2, 0, "zipkin-jdbc")
+
+    val trace = ApplyTimestampAndDuration(List(
+      Span(20L, "get", 20L,
+        annotations = List(
+          Annotation(today * 1000, Constants.ServerRecv, Some(zipkinWeb)),
+          Annotation((today + 350) * 1000, Constants.ServerSend, Some(zipkinWeb)))),
+      Span(20L, "call", 21L, Some(20L),
+        binaryAnnotations = List(
+          BinaryAnnotation(Constants.LocalComponent, "depth2", Some(zipkinWeb)))),
```

Comment:
I think I understand, though maybe not share that capabilities api is
always the answer to this problem. At the moment, this repo is the master,
and implementations follow that or skip test.

Divergence within zipkin has so far been transient, ex people not making
time to adjust leaf implementations.

Capabilities api is a better solution for things that are relatively
permanent, and notable enough to justify the complexity of such an api, and
moreover actionable outside tests (ex users actually invoke the api).

It isn't even needed to share these scala interfaces, it is just a
convenience. Ex java doesn't use these at all except to run tests.
Similarly if you wrote a python dependency store you wouldn't be calling a
capabilities api written in scala.

My 2p is I wouldn't write a capabilities api only to drive tests.. Ignoring
tests that don't pass and keeping issues or README is more appropriate and
better use of the limited time.

IOTW, there's nothing about this change that leads directly to a
capabilities api. We may just feel differently about that.